### PR TITLE
Rename "interpolation" to "evaluation" in SplineEvaluator(2D)

### PR DESCRIPTION
**Pull request by blegouix**
As discussed in https://github.com/CExA-project/ddc/pull/430#discussion_r1587745350 , we should not use the terminology `interpolation` in Spline evaluators.

### CODE REVIEW

File: include/ddc/kernels/splines/spline_evaluator.hpp [hunk 1]
```diff
@@ -240,28 +240,27 @@ class SplineEvaluator
...
-            if (coord_eval_interpolation < ddc::discrete_space<bsplines_type>().rmin()
-                || coord_eval_interpolation > ddc::discrete_space<bsplines_type>().rmax()) {
-                coord_eval_interpolation -= Kokkos::floor(
-                                                    (coord_eval_interpolation
-                                                     - ddc::discrete_space<bsplines_type>().rmin())
-                                                    / ddc::discrete_space<bsplines_type>().length())
-                                            * ddc::discrete_space<bsplines_type>().length();
+            if (coord_eval_evaluation < ddc::discrete_space<bsplines_type>().rmin()
+                || coord_eval_evaluation > ddc::discrete_space<bsplines_type>().rmax()) {
+                coord_eval_evaluation -= Kokkos::floor(
```
Comment (EmilyBourne) [inline include/ddc/kernels/splines/spline_evaluator.hpp:249]: `coord_eval_evaluation` Do we need `eval` twice? Maybe `coord_eval_1d`?

File: include/ddc/kernels/splines/spline_evaluator_2d.hpp [hunk 2]
```diff
@@ -740,28 +738,24 @@ class SplineEvaluator2D
...
+        ddc::Coordinate<typename evaluation_mesh_type1::continuous_dimension_type>
+                coord_eval_evaluation1
+                = ddc::select<typename evaluation_mesh_type1::continuous_dimension_type>(
                         coord_eval);
-        ddc::Coordinate<typename interpolation_mesh_type2::continuous_dimension_type>
-                coord_eval_interpolation2
-                = ddc::select<typename interpolation_mesh_type2::continuous_dimension_type>(
+        ddc::Coordinate<typename evaluation_mesh_type2::continuous_dimension_type>
+                coord_eval_evaluation2
+                = ddc::select<typename evaluation_mesh_type2::continuous_dimension_type>(
```
Comment (EmilyBourne) [inline include/ddc/kernels/splines/spline_evaluator_2d.hpp:747]: Similar to above `eval_eval`

Comment (blegouix) [reply include/ddc/kernels/splines/spline_evaluator_2d.hpp]: I renamed for `coord_eval_interest1`

**Review by EmilyBourne — APPROVED** [review]

**Review by tpadioleau — APPROVED** [review]
Seems fine to me

**Pull request merged**